### PR TITLE
Return invites and members property for teams

### PR DESCRIPTION
**Pull request by imcaizheng**
After changes the response from `GET /taas-teams` would look like:

``` json
[
    {
        "id": 16819,
        "name": "Taas project saar project2",
        "invites": [],
        "members": [
            {
                "lastName": "Dev",
                "role": "manager",
                "updatedBy": 40158997,
                "handle": "TCConManager",
                "userId": 40158997,
                "deletedBy": null,
                "createdAt": "2020-12-23T07:14:19.654Z",
                "firstName": "Connect Manager",
                "createdBy": 40158997,
                "isPrimary": false,
                "id": 13714,
                "projectId": 16819,
                "email": "topcoderconnect+Manager@gmail.com",
                "updatedAt": "2020-12-23T07:14:19.655Z"
            },
            {
                "lastName": "Connect",
                "role": "customer",
                "updatedBy": 40158993,
                "handle": "topcoderconnect",
                "userId": 40158993,
                "deletedBy": null,
                "createdAt": "2020-12-23T07:10:22.672Z",
                "firstName": "Topcoder",
                "createdBy": 40158993,
                "isPrimary": true,
                "id": 13713,
                "projectId": 16819,
                "email": "topcoderconnect@gmail.com",
                "updatedAt": "2020-12-23T07:10:22.679Z"
            },
            {
                "lastName": "A",
                "role": "manager",
                "updatedBy": 40158331,
                "handle": "lakshmiathreya",
                "userId": 40158331,
                "deletedBy": null,
                "createdAt": "2021-01-31T17:13:11.975Z",
                "firstName": "Lakshmi",
                "deletedAt": null,
                "createdBy": 40158331,
                "isPrimary": false,
                "id": 13886,
                "projectId": 16819,
                "email": "lakshmi.athreya+ssotest@gmail.com",
                "updatedAt": "2021-01-31T17:13:11.976Z"
            }
        ],
        "weeklyCost": 0,
        "resources": []
    }
]
```

and the response from `GET /taas-teams/:id` would look like:

``` json
{
    "id": 16705,
    "name": "ABC integrtation2",
    "invites": [],
    "members": [
        {
            "lastName": "Manage",
            "role": "manager",
            "updatedBy": 40152856,
            "handle": "pshah_manager",
            "userId": 40152856,
            "deletedBy": null,
            "createdAt": "2020-11-18T15:52:20.421Z",
            "firstName": "Partha",
            "createdBy": 40152856,
            "isPrimary": true,
            "id": 13568,
            "projectId": 16705,
            "email": "vikas.agarwal+pshah_manager@topcoder.com",
            "updatedAt": "2020-11-18T15:52:20.429Z"
        }
    ],
    "weeklyCost": 0,
    "resources": []
}
```

### CODE REVIEW

File: docs/swagger.yaml [hunk 1]
```diff
@@ -2094,6 +2094,18 @@ components:
           type: string
           example: "1212"
           description: "The team name."
+        invites:
+          type: array
+          items:
+            type: object
+          description: "The invites of the project"
+          example: [{"userIds": [40152855], "role": "copilot"}]
```
Comment (maxceem) [inline docs/swagger.yaml:2102]: Example data looks like doesn't follow response:

image

File: docs/swagger.yaml [hunk 2]
```diff
@@ -2177,6 +2189,18 @@ components:
...
```
Comment (maxceem) [inline docs/swagger.yaml:2197]: Example data looks like doesn't follow response:

image

File: docs/swagger.yaml [hunk 3]
```diff
@@ -2177,6 +2189,18 @@ components:
...
+        members:
+          type: array
+          items:
+            type: object
+          description: "The members of the project"
+          example: [{"lastName": "J", "role": "manager", "updatedBy": 8547899, "handle": "TonyJ", "userId": 8547899, "deletedBy": null, "createdAt": "2021-01-12T04:34:15.006Z", "firstName": "Tony", "createdBy": 8547899, "isPrimary": true, "id": 13819, "projectId": 16893, "email": "ajefts@topcoder.com", "updatedAt": "2021-01-12T04:34:15.018Z"}]
```
Comment (maxceem) [inline docs/swagger.yaml:2203]: Let's replace the real email `ajefts@topcoder.com` with some dummy one to avoid unnecessary exposure?

**Review by maxceem — CHANGES_REQUESTED** [review]
Thanks for the quick PR on the weekend @imcaizheng.

There are a few comments on the code.

Also, I never get `invites`, but I believe this is the issue with M2M token which doesn't have the correct scope. As soon I get the M2M config updated I would tests again.

**Comment by imcaizheng** [comment]
@maxceem The example values for invites and members fixed.

**Review by maxceem — APPROVED** [review]
Works good now @imcaizheng.

We've also added the scope `read:project-invites` to the M2M config, so invites are returned now.

**Pull request merged**